### PR TITLE
Various improvements to tests

### PR DESCRIPTION
**Pull request by rneugeba**
- Some `qemu` tests did not clean up after themselves
- Explicitly test that qemu runs in a container if there is a local install
- Run the kernel config tests on all supported Kernels, not just 4.9
- Add an initial test for hyperkit
- Fix the qemu test for EFI firmware. It didn't work on macOS

### CODE REVIEW

File: test/cases/010_platforms/000_qemu/100_container/test.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  image: "linuxkit/kernel:4.9.x"
+  cmdline: "console=ttyS0 console=tty0 page_poison=1"
+init:
+  - linuxkit/init:f71c3b30ac1ba4ef16c160c89610fa4976f9752f
+  - linuxkit/runc:b0fb122e10dbb7e4e45115177a61a3f8d68c19a9
+  - linuxkit/containerd:60e2486a74c665ba4df57e561729aec20758daed
+onboot:
+  - name: poweroff
+    image: "linuxkit/poweroff:961412b8ef5c5285de0d40ec076701d955eaa084"
+    pid: host
```
Comment (justincormack) [inline test/cases/010_platforms/000_qemu/100_container/test.yml]: you dont need pid: host and caps any more, its in the label

File: test/cases/020_kernel/000_config_4.4.x/test-kernel-config.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+onboot:
+  - name: check-kernel-config
+    image: "linuxkit/test-kernel-config:ecff41279ccbc408079a3996a956432651c6eb9c"
+    readonly: true
+  - name: poweroff
+    image: "linuxkit/poweroff:961412b8ef5c5285de0d40ec076701d955eaa084"
+    pid: host
+    command: ["/bin/sh", "/poweroff.sh", "3"]
+    capabilities:
+     - CAP_SYS_BOOT
```
Comment (justincormack) [inline test/cases/020_kernel/000_config_4.4.x/test-kernel-config.yml]: same here

File: test/cases/020_kernel/002_config_4.10.x/test-kernel-config.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+onboot:
+  - name: check-kernel-config
+    image: "linuxkit/test-kernel-config:ecff41279ccbc408079a3996a956432651c6eb9c"
+    readonly: true
+  - name: poweroff
+    image: "linuxkit/poweroff:961412b8ef5c5285de0d40ec076701d955eaa084"
+    pid: host
+    command: ["/bin/sh", "/poweroff.sh", "3"]
+    capabilities:
+     - CAP_SYS_BOOT
```
Comment (justincormack) [inline test/cases/020_kernel/002_config_4.10.x/test-kernel-config.yml]: and here

**Comment by justincormack** [comment]
Seem to be ssh issues in CI with remote host identification...

**Comment by rneugeba** [comment]
actually, the new qemu in a container test failed. pushed a fix (hopefully)

**Pull request merged**